driver-sync/src/test/functional/com/mongodb/client/unified/ErrorMatcher.java

```diff
@@ -56,10 +57,10 @@ void assertErrorsMatch(final BsonDocument expectedError, final Exception e) {
         }
         if (expectedError.containsKey("isClientError")) {
             assertEquals(context.getMessage("Exception must be of type MongoClientException or IllegalArgumentException"
-                            + " or IllegalStateException or MongoSocketException"),
+                            + " or IllegalStateException or MongoSocketException or MongoInternalException"),
                     expectedError.getBoolean("isClientError").getValue(),
                     e instanceof MongoClientException || e instanceof IllegalArgumentException || e instanceof IllegalStateException
-                            || e instanceof MongoSocketException);
+                            || e instanceof MongoSocketException || e instanceof MongoInternalException);
```

Comment:
Curious where the MongoInternalException is being thrown from, and why.

Reply:
Reverted - was needed due to not using the SyncClientEncryption